src/lm.jl

```diff
@@ -156,7 +156,30 @@ function coeftable(mm::LinearModel)
               ["x$i" for i = 1:size(mm.pp.X, 2)], 4)
 end
 
-predict(mm::LinearModel, newx::Matrix) =  newx * coef(mm)
+# Different methods of prediction for the different factorization types
+
+predict(mm::LinearModel, newx::AbstractMatrix) = newx * coef(mm)
+
+function predict(mm::LinearModel, newx::AbstractMatrix, interval_type::Symbol, level = 0.95)
+    prediction = newx * coef(mm)
+    interval_type == :confint || error("only :confint is currently implemented") #:predint will be implemented
+
+    R = cholfact!(mm.pp)[:U] #get the R matrix from the QR factorization
+    residvar = (ones(3,1) * deviance(mm)/dof_residual(mm))'
+
+    function _local(x::StridedVector)
+        Ac_ldiv_B!(R,x)
+        residvar * (x⋅x)
+    end
+
+    interval = similar(prediction)
+    for i in 1:length(interval)
+        interval[i] = _local(newx[i,:])[1]
+    end
+
+    interval = quantile(TDist(dof_residual(mm)), (1 - level)/2) * sqrt(interval)
```

Comment:
Here is a version very close to what you suggested, I think, but that is slower:
```julia
function predict2(mm::LinearModel, x::StridedVector, interval_type::Symbol, level = 0.95)
    prediction = x' * coef(mm)
    interval_type == :confint || error("only :confint is currently implemented") #:predint will be implemented

    R = cholfact!(mm.pp)[:U] #get the R matrix from the QR factorization
    residvar = deviance(mm)/dof_residual(mm)
    tvalue = quantile(TDist(dof_residual(mm)), (1 - level)/2)

    Ac_ldiv_B!(R,x)
    tmp = residvar * (x⋅x) # ready for modifying if I need a prediction interval
    interval = tvalue * sqrt(tmp)

    prediction[1], (prediction .+ interval)[1], (prediction .- interval)[1]
end

function predict2(mm::LinearModel, newx::AbstractMatrix, interval_type::Symbol, level = 0.95)
    y = similar(newx[:,1])
    yl, yh = similar(y), similar(y)

    for i in 1:size(y,1)
        y[i], yh[i], yl[i] = predict2(mm, newx[i,:], interval_type, level)
    end

    y, yh, yl
end
```
![skaermbillede 2016-12-16 kl 14 55 05](https://cloud.githubusercontent.com/assets/8429802/21265027/b349e2f2-c39f-11e6-851d-4f9e40291bc1.png)